src/getExtendedPublicKey.c

```diff
@@ -0,0 +1,143 @@
+#include <os_io_seproxyhal.h>
+#include <stdint.h>
+
+#include "assert.h"
+#include "os.h"
+#include "errors.h"
+
+#define MAX_BIP32_PATH 10
+#define BIP_44 44
+#define ADA_COIN_TYPE 1815
+#define ADA_ACCOUNT 0
+#define HARDENED_BIP32 0x80000000
+
+#define P1_RECOVERY_PASSPHRASE 0x01
+#define P1_ADDRESS_PUB_KEY 0x02
+
+uint32_t bip32Path[MAX_BIP32_PATH];
+uint8_t pathLength;
+uint8_t privateKeyData[64];
+uint8_t chainCode[32];
+cx_ecfp_public_key_t publicKey;
+cx_ecfp_256_extended_private_key_t privateKey;
+// cx_ecfp_private_key_t privateKey;
+
+void io_exchange_address();
+
+void derive_bip32_node_private_key()
+{
+	STATIC_ASSERT(CX_APILEVEL >= 5, unsupported_api_level);
+	os_memset(chainCode, 0, sizeof(chainCode));
+
+	os_perso_derive_node_bip32(
+	        CX_CURVE_Ed25519,
+	        bip32Path,
+	        pathLength,
+	        privateKeyData,
+	        chainCode);
+
+	cx_ecfp_init_private_key(
+	        CX_CURVE_Ed25519,
+	        privateKeyData,
+	        32,
+	        &privateKey);
+}
+
+void ensureParametersAreCorrect(
+        uint8_t p1,
+        uint8_t p2,
+        uint8_t *dataBuffer,
+        uint16_t dataLength)
+{
+	if (!( p1 == 0 &&
+	       p2 == 0 &&
+	       dataLength >= 1 &&
+	       dataLength == dataBuffer[0] * 4 + 1 )) {
+		THROW(SW_INVALID_REQUEST_PARAMETERS);
+	}
+}
+
+void initializePath(uint8_t *dataBuffer)
+{
+	pathLength = dataBuffer[0];
+
+	uint8_t i = 0;
+	for (i = 0; i < pathLength; i++) {
+		uint8_t offset = 1 + 4 * i;
+
+		bip32Path[i] = (dataBuffer[0 + offset] << 24) |
```

Comment:
there is U4BE macro defined somewhere in `os.h` which reads `uint32_t` from big endian